addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/radiora/handler/RS232Handler.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lutron.internal.radiora.handler;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.lutron.internal.radiora.RadioRAConnection;
+import org.openhab.binding.lutron.internal.radiora.RadioRAFeedbackListener;
+import org.openhab.binding.lutron.internal.radiora.RS232Connection;
+import org.openhab.binding.lutron.internal.radiora.config.RS232Config;
+import org.openhab.binding.lutron.internal.radiora.protocol.RadioRACommand;
+import org.openhab.binding.lutron.internal.radiora.protocol.RadioRAFeedback;
+import org.openhab.binding.lutron.internal.radiora.protocol.ZoneMapInquiryCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RS232Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jeff Lauterbach - Initial contribution
+ */
+public class RS232Handler extends BaseBridgeHandler implements RadioRAFeedbackListener {
+
+    private Logger logger = LoggerFactory.getLogger(RS232Handler.class);
+
+    private RadioRAConnection chronosConnection;
+
+    private RS232Config config;
+    private ScheduledFuture<?> zoneMapScheduledTask;
+
+    RS232Handler(Bridge bridge, RadioRAConnection connection) {
+        super(bridge);
+
+        this.chronosConnection = connection;
+    }
+
+    public RS232Handler(Bridge bridge) {
+        super(bridge);
+
+        this.chronosConnection = new RS232Connection();
+        this.chronosConnection.setListener(this);
+
+    }
+
+    @Override
+    public void dispose() {
+        if (zoneMapScheduledTask != null) {
+            zoneMapScheduledTask.cancel(true);
```

Comment:
Can you cancel a cancelled job / should you check or null it after canceling?

The same question below.

Reply:
The cancel method will return false if it was already cancelled.

Reply:
Hmm I misinterpreted the sentence in the javadoc: 'This attempt will fail if the task has already completed, has already been cancelled, or could not be cancelled for some other reason.' To me failing is not the same as '`false` if the task could not be cancelled, typically because it has already completed normally; `true` otherwise' 

But to summarize, you are right (Y)